### PR TITLE
Fixed KDE Plot to drop the missing values

### PR DESCRIPTION
**Pull request by keshavramaswamy**
- [x] closes https://github.com/pandas-dev/pandas/issues/14821
 - [x] tests added / passed
 - [ ] passes ``git diff upstream/master | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Can you add a test and a release notes in whatsnew/0.19.2?

**Comment by keshavramaswamy** [comment]
Yes, I can do that

**Comment by jreback** [comment]
can you rebase, problem with appeveyor which i just fixed

**Comment by keshavramaswamy** [comment]
Sure, I'll do that!

File: doc/source/whatsnew/v0.19.2.txt [hunk 1]
```diff
@@ -78,4 +78,8 @@ Bug Fixes
 
 - Explicit check in ``to_stata`` and ``StataWriter`` for out-of-range values when writing doubles (:issue:`14618`)
 
+<<<<<<< HEAD
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.19.2.txt]: You can remove the git conflict markers.

**Review by TomAugspurger — CHANGES_REQUESTED** [review]
Still need a test too.  That can go in pandas/tests/plotting/test_series.py There's already a test in there with missing values `test_kde_missing_vals`, apparently not the right mix of missing values though.

Put a comment with a link to this issue when you make the new test. Thanks!

File: doc/source/whatsnew/v0.19.2.txt (hunk 1 above)
Comment (keshavramaswamy) [inline doc/source/whatsnew/v0.19.2.txt]: yes, I noticed that - will add the required test too asap!

**Comment by keshavramaswamy** [comment]
@jreback - Is there a problem with the Travis CI?

**Comment by jreback** [comment]
you need to rebase on current master, I fixed this yesterday.

**Comment by keshavramaswamy** [comment]
Thanks @jreback - have rebased now.

File: pandas/tests/plotting/test_series.py [hunk 2]
```diff
@@ -569,7 +569,11 @@ def test_kde_missing_vals(self):
         _skip_if_no_scipy_gaussian_kde()
         s = Series(np.random.uniform(size=50))
         s[0] = np.nan
-        _check_plot_works(s.plot.kde)
+        axes = _check_plot_works(s.plot.kde)
+        # check if the values have any missing values
+        # GH14821
+        self.assertTrue(any(~np.isnan(axes.lines[0]._xorig)),
```
Comment (TomAugspurger) [inline pandas/tests/plotting/test_series.py]: Can you replace the `._xorig` with `.get_xdata()`? Better to not use matplotlib's internal APIs.

**Comment by TomAugspurger** [comment]
One small change, then we should be good! We're aware of that travis failure on the 3.4 build.

**Review by TomAugspurger — APPROVED** [review]

**Comment by TomAugspurger** [comment]
@keshavramaswamy thanks!

**Pull request merged**